internal/etag/reader.go

```diff
@@ -153,3 +167,41 @@ type VerifyError struct {
 func (v VerifyError) Error() string {
 	return fmt.Sprintf("etag: expected ETag %q does not match computed ETag %q", v.Expected, v.Computed)
 }
+
+// UUIDHash - use uuid to make md5sum
+type UUIDHash struct {
+	uuid string
+}
+
+// Write -  implement hash.Hash Write
+func (u UUIDHash) Write(p []byte) (n int, err error) {
+	return len(p), nil
+}
+
+// Sum -  implement md5.Sum
+func (u UUIDHash) Sum(b []byte) []byte {
+	hasher := md5.New()
+	return hasher.Sum([]byte(u.uuid))
+}
+
+// Reset -  implement hash.Hash Reset
+func (u UUIDHash) Reset() {
+	return
+}
+
+// Size -  implement hash.Hash Size
+func (u UUIDHash) Size() int {
+	return md5.Size
```

Comment:
You are returning `len(u.uuid)` in Sum(...), which is the encoded length, not the raw length.

But you probably want to return the raw value anyway.

Reply:
Not really. md5.Size is a const values which is the encoded length @klauspost